# Gender Analysis - Intercoder Reliability 

Reading Gender coding files for each group 

Packages

In [205]:
pip install datacompy 
import pandas as pd
#DataComPy library is a package to compare two Pandas DataFrames and provide a human-readable output describing the differences.


Note: you may need to restart the kernel to use updated packages.


In [219]:
df1_1=pd.read_csv('G1-1.csv')
df1_2=pd.read_csv('G1-2.csv')
df2_1=pd.read_csv('G2-1.csv')
df2_2=pd.read_csv('G2-2.csv')
df3_1=pd.read_csv('G3-1.csv')
df3_2=pd.read_csv('G3-2.csv')
df4_1=pd.read_csv('G4-1.csv')
df4_2=pd.read_csv('G4-2.csv')

### Group1 (Janet1 - Daniela2)
Number of posts in common: All

In [220]:
import datacompy
compare = datacompy.Compare(
    df1_1,
    df1_2,
    join_columns='SAMPLED CONTENT (URL to content)', 
)
print(compare.report())

DataComPy Comparison
--------------------

DataFrame Summary
-----------------

  DataFrame  Columns  Rows
0       df1       25   159
1       df2       25   159

Column Summary
--------------

Number of columns in common: 25
Number of columns in df1 but not in df2: 0
Number of columns in df2 but not in df1: 0

Row Summary
-----------

Matched on: sampled content (url to content)
Any duplicates on match values: No
Absolute Tolerance: 0
Relative Tolerance: 0
Number of rows in common: 159
Number of rows in df1 but not in df2: 0
Number of rows in df2 but not in df1: 0

Number of rows with some compared columns unequal: 120
Number of rows with all compared columns equal: 39

Column Comparison
-----------------

Number of columns compared with some values unequal: 23
Number of columns compared with all values equal: 2
Total number of values which compare unequal: 382

Columns with Unequal Values or Types
------------------------------------

                          Column df1 dtype df2 dty

### Group2 (Jenny1 - Nemanja2)
Number of posts in common: All

In [221]:
import datacompy
compare = datacompy.Compare(
    df2_1,
    df2_2,
    join_columns='SAMPLED CONTENT (URL to content)', 
)
print(compare.report())

DataComPy Comparison
--------------------

DataFrame Summary
-----------------

  DataFrame  Columns  Rows
0       df1       25   158
1       df2       25   158

Column Summary
--------------

Number of columns in common: 25
Number of columns in df1 but not in df2: 0
Number of columns in df2 but not in df1: 0

Row Summary
-----------

Matched on: sampled content (url to content)
Any duplicates on match values: No
Absolute Tolerance: 0
Relative Tolerance: 0
Number of rows in common: 158
Number of rows in df1 but not in df2: 0
Number of rows in df2 but not in df1: 0

Number of rows with some compared columns unequal: 129
Number of rows with all compared columns equal: 29

Column Comparison
-----------------

Number of columns compared with some values unequal: 20
Number of columns compared with all values equal: 5
Total number of values which compare unequal: 359

Columns with Unequal Values or Types
------------------------------------

                          Column df1 dtype df2 dty

### Group3 (Rodrigo1 - Alaa2)
Number of posts in common: 46 out of 158

In [222]:
import datacompy
compare = datacompy.Compare(
    df3_1,
    df3_2,
    join_columns='SAMPLED CONTENT (URL to content)', 
)
print(compare.report())

DataComPy Comparison
--------------------

DataFrame Summary
-----------------

  DataFrame  Columns  Rows
0       df1       25   158
1       df2       25   158

Column Summary
--------------

Number of columns in common: 25
Number of columns in df1 but not in df2: 0
Number of columns in df2 but not in df1: 0

Row Summary
-----------

Matched on: sampled content (url to content)
Any duplicates on match values: No
Absolute Tolerance: 0
Relative Tolerance: 0
Number of rows in common: 46
Number of rows in df1 but not in df2: 112
Number of rows in df2 but not in df1: 112

Number of rows with some compared columns unequal: 38
Number of rows with all compared columns equal: 8

Column Comparison
-----------------

Number of columns compared with some values unequal: 20
Number of columns compared with all values equal: 5
Total number of values which compare unequal: 157

Columns with Unequal Values or Types
------------------------------------

                          Column df1 dtype df2 dt

### Group4 (Carolina1 - Sunyoung2)
Number of posts in common: All

In [223]:
import datacompy
compare = datacompy.Compare(
    df4_1,
    df4_2,
    join_columns='SAMPLED CONTENT (URL to content)', 
)
print(compare.report())

DataComPy Comparison
--------------------

DataFrame Summary
-----------------

  DataFrame  Columns  Rows
0       df1       25   156
1       df2       25   156

Column Summary
--------------

Number of columns in common: 25
Number of columns in df1 but not in df2: 0
Number of columns in df2 but not in df1: 0

Row Summary
-----------

Matched on: sampled content (url to content)
Any duplicates on match values: No
Absolute Tolerance: 0
Relative Tolerance: 0
Number of rows in common: 155
Number of rows in df1 but not in df2: 1
Number of rows in df2 but not in df1: 1

Number of rows with some compared columns unequal: 135
Number of rows with all compared columns equal: 20

Column Comparison
-----------------

Number of columns compared with some values unequal: 23
Number of columns compared with all values equal: 2
Total number of values which compare unequal: 483

Columns with Unequal Values or Types
------------------------------------

                          Column df1 dtype df2 dty